### ISSUE TITLE
Bug with element taking too long to be hidden

### ISSUE THREAD

**Issue by batata004** [Semantic-Org/Semantic-UI#5654]
Hi,

This bug is easily reproduceable. It happens like this: if you set visibility:hidden to a ".reveal" element it takes almost 1 to 2 seconds to actually hide any image inside the .reveal element.

See it here -> http://quemfazsite.com.br/temp/teste3.php

Click the phrase  "CLICK HERE TO MAKE HIDDEN" and you will see that the #teste element will be hidden immediatly (as it should) BUT for some reason the image INSIDE #teste takes more 1 or 2 seconds to actually disappears.

This bug may seen unharmful at first glance but it's causing serious design problems in these 2 sites:

https://www.federalsc.com.br
https://www.deobraemobra.com.br

Open any of these websites and scroll down. I use a very simple animate plugin that hides content and when the user scrolls the content appears. The problem happening here is because hiding the content is taking too long (because of the bug) and when the script animates the content to be shown (with flip effect) it is not beautifull cause the image inside the reveal element is not being hidden fastly enough.

If you need any additional info about this bug I am here to help but if you open the first link in this thread showing the bug with the minimmal html necessary to run it, you will see clearly the bug happening.

**Comment by Banandrew** [comment]
Hi @batata004, again, not a bug, that’s expected behavior, because of [this](https://github.com/Semantic-Org/Semantic-UI/blob/master/dist/components/reveal.css#L27). If you don’t want it, [override it](https://github.com/Semantic-Org/Semantic-UI/blob/master/src/themes/default/elements/reveal.variables#L8); something like `transition: opacity .5s ease .1s;` should work.

**Comment by batata004** [comment]
My dear friend @Banandrew  I am really sorry for taking so much of your time with my problems that I think are bugs. Anyway, I tried understanding what you said in the last comment but I didnt get it.

I must override the reveal class, right? But I want to keep the reveal exactly as it works now (it has a nice animation) but I also would like to be able to set it to hidden (visibility) instantly. Is it possible?

**Comment by Banandrew** [comment]
@batata004 No, you need to override the variable that’s being used by the reveal, and it will work exactly the same. I think it’ll be easier to understand if [you read about the transition rule](http://devdocs.io/css/transition)—instead of transitioning all properties, we only transition opacity; open `installation_folder/src/site/elements/reveal.variables`, copy `@transition: opacity @transitionDuration @defaultEasing @transitionDelay;` there, then rebuild the framework.

**Comment by batata004** [comment]
@Banandrew my friend, great link you provided me! It taught me a lot!

So let me just be clear: I checked all my styles and none of them are transitioning all the properties. They only transition what is needed. In the link http://quemfazsite.com.br/temp/teste3.php I never used "all" in the transition. So this problem shouldnt be happening, right?

Maybe a practical solution would help me understand this, if you have some time :) See my friend, I always use CDN to import semantic ui, I use it in many websites and storing a local copy in every website is not good. I use a CDN so I cant override the framework.

So my question is this: is there any class/property/css that I can copy/paste to the link below in order to solve this problem?  I tried adding .reveal{transition: opacity .5s ease .1s;} to the link below but it didnt work. ;(

http://quemfazsite.com.br/temp/teste3.php

**Comment by Banandrew** [comment]
@batata004 

>So let me just be clear: I checked all my styles and none of them are transitioning all the properties. They only transition what is needed. In the link http://quemfazsite.com.br/temp/teste3.php I never used "all" in the transition. So this problem shouldnt be happening, right?

If you’ll inspect `.visible.content`, you’ll see that it has a `transition` property of `all .5s ease .1s` that I mentioned in the first post.

>Maybe a practical solution would help me understand this, if you have some time :) See my friend, I always use CDN to import semantic ui, I use it in many websites and storing a local copy in every website is not good. I use a CDN so I cant override the framework.

Just to elaborate on this, it works right now, but it can quickly become cumbersome—that’s why people invented better ways to write CSS, preprocessors and whatnot, so I’m highly suggesting to stop using CDNs if you often need to override a lot of things.

>So my question is this: is there any class/property/css that I can copy/paste to the link below in order to solve this problem? I tried adding .reveal{transition: opacity .5s ease .1s;} to the link below but it didnt work.

Sure, just add `transition: opacity .5s ease .1s;` to `.visible.content`.

**Comment by batata004** [comment]
I know this thread is already closed and I just want to contribute with the community: to solve the delay problem with the visibility taking too long to happen, I just changed my code to use opacity instead.

I mean: intestad of using visibility:hidden and visibility:visible I changed to opacity:0 and opacity:1 respectively. The problem went away and the element animates IMMEDIATLY!

Hope this help people trying to use the .reveal element and is experiencing too long time to it to get hidden.
